logstreams/src/main/java/io/zeebe/distributedlog/impl/DefaultDistributedLogstreamService.java

```diff
@@ -236,19 +271,35 @@ public void backup(BackupOutput backupOutput) {
     // backup snapshot, but not appended to logStorage may be lost. So, if this node is away for a
     // while and tries to recover with backup received from other nodes, there will be missing
     // entries in the logStorage.
-    logger.info("Backup log {} at position {}", logName, lastPosition);
+    logger.info(
+        "Backup log {} at position {}, last append index {}",
+        logName,
+        lastPosition,
+        lastAppendIndex);
 
     // Backup in-memory states
     backupOutput.writeLong(lastPosition);
+    backupOutput.writeLong(lastAppendIndex);
     backupOutput.writeString(currentLeader);
     backupOutput.writeLong(currentLeaderTerm);
   }
 
   @Override
   public void restore(BackupInput backupInput) {
+    LOG.debug("Restoring primitive of partition {}", partitionId);
     final long backupPosition = backupInput.readLong();
+    final long backLastAppendIndex = backupInput.readLong();
+    lastAppendIndex = backLastAppendIndex;
 
     if (lastPosition < backupPosition) {
+      // We had a fail over or the follower just lags behind.
```

Comment:
A minor thing: failover doesn't cause a restore.

Reply:
Sure? As we discussed today? Follower already received the snapshot but didnt reached the index yet?

Reply:
If the follower is fast enough to apply the operations, then it never have to install the snapshot. The case we discussed today is when the follower is slow in applying and then it receives a snapshot. In that case, it will always install the snapshot whether a fail-over happens or not.

Reply:
so it just when someone lags behind :D

Reply:
Yes :smile: